x-pack/solutions/search/plugins/search_connectors/server/services/index.ts

```diff
@@ -234,20 +253,23 @@ export class AgentlessConnectorsInfraService {
   };
 }
 
-export const getConnectorsWithoutPolicies = (
+export const getConnectorsToDeploy = (
   packagePolicies: PackagePolicyMetadata[],
   connectors: ConnectorMetadata[]
 ): ConnectorMetadata[] => {
   return connectors.filter(
-    (x) => packagePolicies.filter((y) => y.connector_metadata.id === x.id).length === 0
+    (x) =>
+      packagePolicies.filter((y) => y.connector_settings.id === x.id).length === 0 &&
```

Comment:
nit: I once got linked this on my PR [Cognitive load is what matters: complex conditionals](https://github.com/zakirullin/cognitive-load?tab=readme-ov-file#complex-conditionals)

I'm struggling to parse this boolean expression in my head, maybe we can split it up:
```
const statementIWithClearName1 = ...
const statementIWithClearName2 = ...
return statementIWithClearName1 && statementIWithClearName2
```

Reply:
maybe more elegant with every lib function: 

```suggestion
      packagePolicies.every((policy) => policy.connector_settings.id !== x.id)
```

Reply:
Lemme see how I can make it simpler